h2o-core/src/main/java/water/fvec/Frame.java

```diff
@@ -1633,19 +1635,24 @@ public int getCurrentRowSize() throws IOException {
       assert av > 0;
       return _line.length;
     }
+    
 
     byte[] getBytesForRow() {
       StringBuilder sb = new StringBuilder();
       BufferedString tmpStr = new BufferedString();
-      for (int i = 0; i < _curChks.length; i++ ) {
+      for (int i = 0; i < _curChks.length; i++) {
         Vec v = _curChks[i]._vec;
-        if(i > 0) sb.append(_parms._separator);
-        if(!_curChks[i].isNA(_chkRow)) {
-          if( v.isCategorical() ) sb.append('"').append(v.factor(_curChks[i].at8(_chkRow))).append('"');
-          else if( v.isUUID() ) sb.append(PrettyPrint.UUID(_curChks[i].at16l(_chkRow), _curChks[i].at16h(_chkRow)));
-          else if( v.isInt() ) sb.append(_curChks[i].at8(_chkRow));
-          else if (v.isString()) sb.append('"').append(_curChks[i].atStr(tmpStr, _chkRow)).append('"');
-          else {
+        if (i > 0) sb.append(_parms._separator);
+        if (!_curChks[i].isNA(_chkRow)) {
+          if (v.isCategorical()) {
+            final String escapedString = escapeQuotesForCsv(v.factor(_curChks[i].at8(_chkRow)));
```

Comment:
Can we avoid escaping the same categorical value each time? Or maybe - since it is unlikely that categoricals will have quote characters at all, can we just have a per-column flag that would say that the domain doesn't have any categoricals that need escaping (the most common case)?

Reply:
1. This very case is recognized by H2O as categorical vector. And it has quotes :smile: 
2. Excellent suggestion. Will do. Thank you.

Reply:
Implemented your suggestion.